### PR TITLE
feat: Improved compliance of URL object

### PR DESCRIPTION
**Pull request by nabetti1720**
### Description of changes

Add the following methods to improve the compliance of `URL` Object.
See also. https://url.spec.whatwg.org/#dom-url-parse

- URL.parse()

### Checklist

- [x] Created unit tests in `tests/unit` and/or in Rust for my feature if needed
- [x] Ran `make fix` to format JS and apply Clippy auto fixes
- [x] Made sure my code didn't add any additional warnings: `make check`
- [x] Updated documentation if needed ([API.md](API.md)/[README.md](README.md)/Other)

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Review by richarddavison — CHANGES_REQUESTED** [review]
Thanks for the PR, just a tiny improvement and I'm happy :)

File: llrt_core/src/modules/http/url.rs [hunk 1]
```diff
@@ -64,31 +64,43 @@ impl<'js> URL<'js> {
...
     #[qjs(static)]
     pub fn can_parse(ctx: Ctx<'js>, input: Value<'js>, base: Opt<Value<'js>>) -> bool {
+        !Self::parse(ctx.clone(), input.clone(), Opt(base.clone())).is_null()
+    }
+
+    #[qjs(static)]
+    pub fn parse(ctx: Ctx<'js>, input: Value<'js>, base: Opt<Value<'js>>) -> Value<'js> {
         if let Some(base) = base.0 {
-            let base_string = match get_string(&ctx, base) {
+            let base_string = match get_string(&ctx.clone(), base) {
```
[the remark this answers is not in the document]
Comment (nabetti1720) [reply llrt_core/src/modules/http/url.rs]: It is a remnant from the days of trial and error. Fixed.

File: llrt_core/src/modules/http/url.rs [hunk 2]
```diff
@@ -64,31 +64,43 @@ impl<'js> URL<'js> {
...
                 Ok(s) => s,
-                Err(_) => return false,
+                Err(_) => return Null.into_js(&ctx).unwrap(),
             };
-            let path_string = match get_string(&ctx, input) {
+            let path_string = match get_string(&ctx.clone(), input) {
                 Ok(s) => s,
-                Err(_) => return false,
+                Err(_) => return Null.into_js(&ctx).unwrap(),
```
[the remark this answers is not in the document]
Comment (nabetti1720) [reply llrt_core/src/modules/http/url.rs]: This has also been corrected. Please review.

**Review by richarddavison — APPROVED** [review]
Awesome thanks!

**Pull request merged**